azure-pipelines.yml

```diff
@@ -0,0 +1,66 @@
+# ASP.NET
+# trigger on main pull
+trigger:
+  branches:
+    include:
+    - main
+# initial build
+stages:
+- stage: build
+  jobs:
+  - job: build
+    variables:
+        imageName: 'fakebookposts'
+
+    steps:
+    - task: Docker@2
+      displayName: Build Docker Image
+      inputs:
+        repository: $(imageName)
+        command: 'build'
+        Dockerfile: 'Fakebook.Posts/Dockerfile'
+# sonar cloud
+- stage: sonarCloud
+  jobs:
+  - job: build
+    pool:
+      vmImage: ubuntu-latest
+    variables:
+      buildConfiguration: 'Release'
+      directory: 'Fakebook.Posts/Fakebook.Posts.WebApi'
+
+    steps:
+    - task: UseDotNet@2
+      displayName: Use sdk 2.x
+      inputs:
+        packageType: sdk
+        version: 2.x
+    - task: SonarCloudPrepare@1
+      inputs:
+        SonarCloud: 'SonarCloud Token'
+        organization: '2011-fakebook-project3'
+        scannerMode: 'MSBuild'
+        projectKey: '2011-fakebook-project3_posts'
+        extraProperties: |
+          sonar.cs.opencover.reportsPaths=$(Agent.TempDirectory)/**/coverage.opencover.xml
+          sonar.cs.vstest.reportsPaths=$(Agent.TempDirectory)/*.trx
```

Comment:
FYI we do want to get this trx -> sonarcloud in the other pipelines that don't have it but it's low priority.